src/s2/s2point.h

```diff
@@ -43,6 +43,9 @@ class S2Point : public Vector3_d {
   using Base = Vector3_d;
   using Base::Base;
 
+  // This is required for building on MSVC
+  S2Point() = default;
```

Comment:
Is this an MSVC bug?  I don't suppose they have a public bug tracker. Is there any link that could be added here?

Reply:
According to the spec, to the best of my ability to parse it, the `using Base::Base` declaration doesn't technically create new contractors in the derived class, but just makes the base class constructors available via overloading. So it's sort of left to the interpretation whether those imported "not quite constructors" count for the purposes of the rule that deletes the default ctor when a non-default ctor is defined.  GCC and Clang interpret it one way and MSVC interprets it another, and so far I can't find anything in the spec that's definitive one way or another.  

Personally I lean towards the MSVC interpretation, because if you add a data member to derived with a default value, and look at the GCC [code generated](https://godbolt.org/z/Gn7arTGsG) without an explicitly declared default ctor, it still includes code to initialize that derived member, even though the name of the ctor is `Derived::Base`.  Further, in my original example on compiler explorer, if you switch to the GCC output, the presence or absence of a `Derived() = default` declaration in the derived class produces the exact same output in terms of assembly.

It seems to me that the primary motivation in the `using Base::Base` declaration here (and in general) seems to be in avoiding having to re-declare a bunch of complex ctor signatures in a base class, not to circumvent the rules handling whether the default ctor is deleted.

So, the spec _may_ be ambiguous, and the presence of  `S2Point() = default;` _should_ be neutral on GCC and Clang and is required on MSVC.

Reply:
OK, so I asked on [Reddit](https://www.reddit.com/r/cpp_questions/comments/18zju2t/interaction_between_imported_base_class/kgi6crt/) and got a response that basically said this was ambiguous in the spec originally, but that the clarification was folded into the C++17 process and in theory should have been retroactively applied to C++11.

It appear that the MSVC compiler _does_ accept the code as written if you tell it the standard is C++17, but not at the current codebase level of C++14.

There are [guidelines](https://learn.microsoft.com/en-us/cpp/overview/how-to-report-a-problem-with-the-visual-cpp-toolset?view=msvc-170) for submitting a bug report for the MSVC toolset, but I'm already pretty far down the rabbit hole, since s2geometry is just a dependency of another project that I want to use as a dependency in my own project.

In order to get s2geometry working on MSVC without waiting on the turnaround on reporting the bug and waiting for a fix there are a couple options...

* Accept this change as is, since it should be innocuous with regard to GCC and Clang
* Wrap the change in a `#if __cplusplus >= 201703L || (defined(_MSC_VER) && _MSC_VER >= 1900)` macro

Additionally I can update the comment linking to either the reddit post or the linked paper.

Reply:
Actually, looking at [this page](https://learn.microsoft.com/en-us/cpp/overview/visual-cpp-language-conformance?view=msvc-170) I suspect that reporting this as a bug isn't likely to get any traction, since the mention of complying with the reworded constructor inheritance paper is explicitly tagged as being something that has to be enabled via the `/std:C++17` argument.

Reply:
Can you put a summary of this in a comment and mention that it will no longer be required with `/std:C++17`?

Reply:
commented.